platform/jvm/BUILD.bazel

```diff
@@ -14,9 +14,8 @@ cc_binary(
     linkopts = [
         "-lm",  # Required to avoid dlopen runtime failures unrelated to rust
         "-lz",  # Link against system zlib library
-        "-Wl,-z,max-page-size=16384",  # enable 16 KB ELF alignment
     ] + select({
-        "@platforms//os:android": ["-Wl,--retain-symbols-file=$(location :jni_symbols.lds)"],
+        "@platforms//os:android": ["-Wl,-z,max-page-size=16384,--retain-symbols-file=$(location :jni_symbols.lds)"],
```

Comment:
this should work, the best way to test this still works would be not to test before/after the pr chances but with and without this flag